packages/client/src/clients/in-app-message/types.ts

```diff
@@ -0,0 +1,103 @@
+import { GenericData, PageInfo } from "@knocklabs/types";
+
+import { NetworkStatus } from "../../networkStatus";
+import { NotificationSource } from "../messages/interfaces";
+
+export interface InAppMessageTextContentField {
+  key: string;
+  rendered: string;
+  type: "text" | "markdown" | "textarea";
+  value: string;
+}
+
+export interface InAppMessageBooleanContentField {
+  key: string;
+  rendered: boolean;
+  type: "boolean";
+  value: boolean;
+}
+
+export interface InAppMessageButtonContentField {
+  key: string;
+  type: "button";
+  text: Exclude<InAppMessageTextContentField, "key | type"> & {
+    key: "text";
+    type: "text";
+  };
+  action: Exclude<InAppMessageTextContentField, "key | type"> & {
+    key: "action";
+    type: "text";
+  };
+}
+
+export type InAppMessageContentField =
+  | InAppMessageTextContentField
+  | InAppMessageBooleanContentField
+  | InAppMessageButtonContentField;
+
+export interface InAppMessage<T = GenericData> {
+  __cursor: string;
+  id: string;
+  message_type: string;
+  schema_variant: string;
+  schema_version: string;
+  content: Record<string, InAppMessageContentField>;
+  data: T | null;
+  inserted_at: string;
+  updated_at: string;
+  seen_at: string | null;
+  read_at: string | null;
+  interacted_at: string | null;
+  archived_at: string | null;
+  link_clicked_at: string | null;
+  source: NotificationSource;
+}
+
+export interface InAppMessageResponse {
+  items: InAppMessage[];
+  pageInfo: PageInfo;
+}
+
+export interface InAppMessagesQueryInfo {
+  networkStatus: NetworkStatus;
+  loading: boolean;
+  data?: InAppMessageResponse;
```

Comment:
Do we need to keep this per-query?

Reply:
Not the full message data, but we at least need the ids so that we can look up the messages under `store.messages`. I'll update the types / store to only store the ids per query